epicure-admin/src/app/components/generic-table/generic-table.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { MatTableDataSource } from '@angular/material/table';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-generic-table',
+  templateUrl: './generic-table.component.html',
+  styleUrls: ['./generic-table.component.scss']
+})
+export class GenericTableComponent implements OnInit {
+  @Input() pageTitle: string = '';
+  @Input() data: any[] = [];
+  @Input() columns: string[] = [];
+  @Input() columnDefs: { [key: string]: string } = {};
+
+  displayedColumns: string[] = [];
+  dataSource = new MatTableDataSource(this.data);
+  
+  constructor(private router: Router) {}
+  ngOnInit() {
+    this.displayedColumns = this.columns;
+    console.log('pageTitle', this.pageTitle);
+    console.log('Columns:', this.columns);
+    console.log('ColumnDefs:', this.columnDefs);
+    console.log('Data:', this.data);
+  }
+  
+  getElementVal(element: any, column: any) {
```

Comment:
This function is messy with a lof of conditions, seperate into functions and simpler conditiononing

Reply:
done